ocaml/libs/tracing/tracing.ml

```diff
@@ -279,6 +279,56 @@ module Spans = struct
 
   let spans = Hashtbl.create 100
 
+  let frequency_tbl = Hashtbl.create 100
+
+  let incr_frequency_tbl ~(span : Span.t) =
+    match Hashtbl.find_opt frequency_tbl span.name with
+    | None ->
+        Hashtbl.add frequency_tbl span.name 1
+    | Some v ->
+        Hashtbl.replace frequency_tbl span.name (succ v)
+
+  let decr_frequency_tbl ~(span : Span.t) =
+    match Hashtbl.find_opt frequency_tbl span.name with
+    | None ->
+        ()
+    | Some v ->
+        if pred v = 0 then
+          Hashtbl.remove frequency_tbl span.name
+        else
+          Hashtbl.replace frequency_tbl span.name (pred v)
+
+  let most_frequent n =
+    let lst =
+      frequency_tbl
+      |> Hashtbl.to_seq
+      |> List.of_seq
+      |> List.sort (fun x y -> compare (snd x) (snd y))
+      |> List.rev
```

Comment:
Sort in reverse order instead:
```
      |> List.sort (fun x y -> compare (snd y) (snd x))
```

Reply:
In general I'm not fond of sorting a list of +1000 elements to get just 10.

I would rather have a dedicated fold over the hashtable to keep the n most frequent values, probably maintained as a Map.

```
let take_most k acc ((k, freq) as elem) =
  let last_min = Map.min_elem acc in
  if freq <= last_min then
    acc
  else
    Map.remove last_min acc |> Map.add freq k
in

let most_frequent = Hashtable.fold (take_most n) Map.empty freq_tbl
```

Reply:
Cool, that's a classic usage. I have kept it in my notes.